src/super_gradients/convert_from_recipe.py

```diff
@@ -0,0 +1,206 @@
+"""
+Entry point for converting recipe file to self-contained train.py file.
+
+General use: python -m super_gradients.convert_from_recipe --config-name="DESIRED_RECIPE".
+For recipe's specific instructions and details refer to the recipe's configuration file in the recipes directory.
+"""
+import argparse
+import collections
+import os.path
+from typing import Tuple, Mapping, Dict
+
+import hydra
+import pkg_resources
+from hydra.core.global_hydra import GlobalHydra
+from omegaconf import DictConfig, OmegaConf, ListConfig
+
+from super_gradients import Trainer
+from super_gradients.common import MultiGPUMode
+from super_gradients.common.abstractions.abstract_logger import get_logger
+from super_gradients.common.environment.omegaconf_utils import register_hydra_resolvers
+from super_gradients.common.environment.path_utils import normalize_path
+from super_gradients.training.utils import get_param
+
+logger = get_logger(__name__)
+
+
+def import_black_or_fail_with_instructions():
+    try:
+        import black
+
+        return black
+    except ImportError:
+        raise ImportError(
+            "Black is not installed. Please install it with `pip install black` and try again. "
+            "If you are using a virtual environment, make sure it is activated."
+        )
+
+
+def import_black_or_install():
+    try:
+        import black
+
+        return black
+    except ImportError:
+        import pip
+
+        pip.main(["install", "black==22.10.0"])
+
+        return import_black_or_fail_with_instructions()
+
+
+black = import_black_or_install()
+
+
+def recursively_walk_and_extract_hydra_targets(cfg, objects=None, prefix=None) -> Tuple[DictConfig, Dict[str, Mapping]]:
+    if objects is None:
+        objects = collections.OrderedDict()
+    if prefix is None:
+        prefix = ""
+
+    if isinstance(cfg, DictConfig):
+        for key, value in cfg.items():
+            value, objects = recursively_walk_and_extract_hydra_targets(value, objects, prefix=f"{prefix}_{key}")
+            cfg[key] = value
+
+        if "_target_" in cfg:
+            target_class = cfg["_target_"]
+            target_params = dict([(k, v) for k, v in cfg.items() if k != "_target_"])
+            object_name = f"{prefix}".replace(".", "_").lower()
```

Comment:
Not sure what the intent was with the prefix and "_", "{index}" stuff....
But seems like it has some problems:

      "valid_metrics_list": [
            {
                "PoseEstimationMetrics": {
                    "num_joints": 17,
                    "oks_sigmas": [
                        0.026,
                        0.025,
                        0.025,
                        0.035,
                        0.035,
                        0.079,
                        0.079,
                        0.072,
                        0.072,
                        0.062,
                        0.062,
                        0.107,
                        0.107,
                        0.087,
                        0.087,
                        0.089,
                        0.089,
                    ],
                    "max_objects_per_image": 30,
                    "post_prediction_callback": _valid_metrics_list_0_poseestimationmetrics_post_prediction_callback,
                }
            }
        ]